tests/test_pair.py

```
@@ -0,0 +1,567 @@
+        self.assertTrue(np.allclose(p22.vals, [1., 0.]))
+
+        # Test from_scalars with None (docstring says None is converted to zero Scalar)
+        p22_none = Pair.from_scalars(1., None)
+        self.assertTrue(np.allclose(p22_none.vals, [1., 0.]))
+
+        p22_none2 = Pair.from_scalars(None, 2.)
+        self.assertTrue(np.allclose(p22_none2.vals, [0., 2.]))
+
+        # Test from_scalars with None and n-D scalars
+        x_nd = Scalar([[1., 2.], [3., 4.]], drank=1)
+        y_nd = Scalar([[5., 6.], [7., 8.]], drank=1)
```

Ruff (0.14.7)

Local variable `y_nd` is assigned to but never used

Remove assignment to unused variable `y_nd`

(F841)